server/src/main/java/au/org/aodn/ogcapi/server/features/RestServices.java

```diff
@@ -39,7 +39,7 @@ public ResponseEntity<Collection> getCollection(String id, String sortBy) throws
             return ResponseEntity.ok()
                     .body(StacToCollection.convert(model.getCollections().get(0)));
         } else {
-            throw new NoSuchElementException(String.format("uuid %s not found!", id));
+            return ResponseEntity.notFound().build();
```

Comment:
can you add more details to the returning message than simply 404 http status code?

Reply:
test also failed

Reply:
👌

Reply:
after double-checking, i found this function want to return: `ResponseEntity<Colleciton>`, so i can only set body whose type is a `Collection`. For not messing up this function, i think only 404 code is enough for this situation, because the client side knows the url and uuid.

Reply:
so cannot do something like this?

```java
return ResponseEntity.status(HttpStatus.NOT_FOUND).body(String.format("uuid %s not found!", id));
```

Reply:
can you please try, otherwise missing out what previously tried to inform.

Reply:
the returning type for the method can be raw `ResponseEntity`

Reply:
or, having a log `log.error(...)`; before returning 404 code.

Reply:
so you don't have to make the controller method returning type using raw `ResponseEntity`.

Reply:
> so cannot do something like this?
> 
> ```java
> return ResponseEntity.status(HttpStatus.NOT_FOUND).body(String.format("uuid %s not found!", id));
> ```

yes this is what i tried, but it only want a `Collection` type for the body 
![image](https://github.com/user-attachments/assets/17ac6f65-8ccf-4147-859c-0c45e0ab84ef)

So i use raw `ResponseEntity` before, not setting body for it.

Reply:
> can you please try, otherwise missing out what previously tried to inform.

the client side is ok, knows the uuid and the NOT_FOUND status. For the server side output, do you think adding a `log.error()` is ok?

Reply:
yep, you can just print the log if too much hassles to deal with, I think the previous impl to get some traces for what happened should stay. I've seen animaltracking controllers always use raw ResponseEntity type for the method declaration.